Improve debug logging to show Slurm node state

enrico-usai opened the pull request:
The output is now something like:
`Current state of Slurm nodes to resume: [('queue1-dy-queue1-t2medium-2', 'ALLOCATED+CLOUD+NOT_RESPONDING+POWERING_UP')]`

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Commit:
Improve logging to show Slurm node state

The output is now something like:
Current state of Slurm nodes to resume: [('queue1-dy-queue1-t2medium-2', 'ALLOCATED+CLOUD+NOT_RESPONDING+POWERING_UP')]
Changed from debug to info.

Signed-off-by: Enrico Usai <usai@amazon.com>

Commit:
Fix typo in SlurmNode method

Signed-off-by: Enrico Usai <usai@amazon.com>

lukeseawalker commented on src/slurm_plugin/resume.py:
```diff
@@ -161,8 +161,12 @@ def _resume(arg_nodes, resume_config):
-    node_list = [node.name for node in get_nodes_info(arg_nodes)]
-    log.debug("Retrieved nodelist: %s", node_list)
+    node_list = []
+    node_list_with_status = []
```
have you evaluated the pros/cons to have just one list?

enrico-usai replied:
The `node_list` is used for `add_instances_for_nodes` by both `clustermgtd.py` and `resume.py`.

In the `clustermgtd.py` the `node_list` is produced in a different way. I prefer to not touch the logic of the `add_instances_for_nodes` just to improve the logging for now. This change can be evaluated separately.

lukeseawalker commented on src/slurm_plugin/resume.py:
```diff
@@ -161,8 +161,12 @@ def _resume(arg_nodes, resume_config):
+    for node in get_nodes_info(arg_nodes):
+        node_list.append(node.name)
+        node_list_with_status.append((node.name, node.state_string))
+    log.debug("Retrieved nodelist: %s", node_list_with_status)
```
do we want this at info level?

demartinofra replied:
+1 to making this info and maybe slightly changing the log message to something like "Current state of Slurm nodes to resume: [node: state, ...]"

enrico-usai replied:
Changed.

Commit:
Upgrade CodeQL GitHub Action to v2

https://github.blog/changelog/2023-01-18-code-scanning-codeql-action-v1-is-now-deprecated/
Signed-off-by: Enrico Usai <usai@amazon.com>
